src/pages/PasswordRecoveryPage/PasswordRecoveryPage.js

```diff
@@ -0,0 +1,185 @@
+// Libraries
+import React, { useEffect, useState } from 'react';
+import NavBar from '../../componentgroups/NavBar_v0.0.1';
+import { Redirect } from 'react-router'
+
+
+// Components
+import FlexBox from '../../components/FlexBox';
+import axios from 'axios';
+
+import LoadingAnime from '../../componentgroups/loading/index'
+
+import './style.css';
+
+
+
+const PasswordRecoveryPage = (props) => {
+
+  const [submitted, setSubmitted] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+  const [successMessage, setSuccessMessage] = useState('');
+  const [redirect, setRedirect] = useState(false);
+  const [password, setPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [passwordMatch, setPasswordMatch] = useState(true);
+  const [validatePassword, setValidatePassword] = useState(true);
+  const [passwordChanged, setPasswordChanged] = useState(false);
+  const [loading, setLoading] = useState(true);
+
+  const [verifyToken, setVerifyToken] = useState(false);
+
+
+  let search = window.location.search;
+  let params = new URLSearchParams(search);
+  let recoveryCode = params.get('reset');
+
+  useEffect((e) => {
+    axios.post("http://localhost:8080/users/forgot-password/verify-token", {
+      recoveryCode: recoveryCode,
+    })
+      .then((res) => {
+        setLoading(false)
+        setVerifyToken(true)
+      })
+      .catch((err) => {
+        setLoading(false)
+        setTimeout(() => setRedirect(true), 3400)
+      })
+  })
+
+
+
+  function handleSubmit(e) {
+    e.preventDefault();
+    setSubmitted(true)
+
+    if (!password || !confirmPassword) {
+      return
+    }
+    const strongRegexPasswordValidation = new RegExp("^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})");
+    if (!strongRegexPasswordValidation.test(password)) {
+      return setValidatePassword(false)
+    }
+
+    if (password != confirmPassword) {
+      setPasswordMatch(false)
+      return
+    }
+
+    setLoading(true)
+    setValidatePassword(true);
+    setPasswordMatch(true);
+    setLoading(true)
+    axios.post('http://localhost:8080/users/forgot-password/recover',
```

Comment:
parameterize this from "localhost:8080". Use a properties file or some such method to retrieve this endpoint.